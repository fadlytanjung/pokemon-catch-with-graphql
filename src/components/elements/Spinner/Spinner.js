import React from 'react';
import PropTypes from 'prop-types';
import styles from './styles.scoped.css';

export default function Spinner(props) {
  const { className, color, size } = props;
  const customClass = [styles.root, className].filter(Boolean).join(' ');
  const style = { height: size, width: size };
  const childStyle = { borderColor: `${color} transparent transparent transparent` };
  // animation from https://loading.io/css/
  return (
    <div className={customClass} style={style}>
      {[...Array.from({ length: 4 }).keys()].map(i => (
        <div key={i} style={childStyle} />
      ))}
    </div>
  );
}

Spinner.defaultProps = {
  className: '',
  color: '#ffffff',
  size: '0.9rem',
};

Spinner.propTypes = {
  className: PropTypes.string,
  color: PropTypes.string,
  size: PropTypes.string,
};
