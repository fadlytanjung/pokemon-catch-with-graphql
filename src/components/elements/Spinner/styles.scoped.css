.root {
  display: inline-block;
  position: relative;
}

.root > div {
  animation: spinner 1.2s cubic-bezier(0.5, 0, 0.5, 1) infinite;
  border: 0.1em solid #ffffff;
  border-color: #ffffff transparent transparent transparent;
  border-radius: 50%;
  height: 80%;
  margin: 10%;
  position: absolute;
  width: 80%;
}

.root > div:nth-child(1) { animation-delay: -0.45s; }
.root > div:nth-child(2) { animation-delay: -0.3s; }
.root > div:nth-child(3) { animation-delay: -0.15s; }

@keyframes spinner {
  0% { transform: rotate(0deg); }
  100% { transform: rotate(360deg); }
}
