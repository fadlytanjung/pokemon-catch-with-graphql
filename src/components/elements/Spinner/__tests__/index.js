import React from 'react';
import ShallowRenderer from 'react-test-renderer/shallow';
import Spinner from '../Spinner';

describe('src/components/elements/Spinner', () => {
  test('render', () => {
    const shallow = new ShallowRenderer();
    const tree = shallow.render(<Spinner />);
    expect(tree).toMatchSnapshot();
  });
});
