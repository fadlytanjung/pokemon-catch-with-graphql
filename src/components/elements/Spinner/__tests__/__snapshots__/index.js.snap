// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src/components/elements/Spinner render 1`] = `
<div
  className=""
  style={
    Object {
      "height": "0.9rem",
      "width": "0.9rem",
    }
  }
>
  <div
    style={
      Object {
        "borderColor": "#ffffff transparent transparent transparent",
      }
    }
  />
  <div
    style={
      Object {
        "borderColor": "#ffffff transparent transparent transparent",
      }
    }
  />
  <div
    style={
      Object {
        "borderColor": "#ffffff transparent transparent transparent",
      }
    }
  />
  <div
    style={
      Object {
        "borderColor": "#ffffff transparent transparent transparent",
      }
    }
  />
</div>
`;
